src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImportedNamespacesValueProvider.cs

```diff
@@ -22,82 +22,86 @@ public ImportedNamespacesValueProvider(ConfiguredProject configuredProject, IPro
         _threadingService = threadingService;
         _projectAccessor = projectAccessor;
         _specialImports = new ConcurrentHashSet<string>();
+        _encoding = new KeyValuePairListEncoding();
     }
 
-    private async Task<ImmutableArray<string>> GetSelectedImportsAsync()
+    private async Task<ImmutableArray<(string Value, bool IsImported)>> GetSelectedImportsAsync()
     {
-        ImmutableArray<string> existingImports = await _projectAccessor.OpenProjectForReadAsync(_configuredProject, project =>
+        ImmutableArray<(string Value, bool IsImported)> existingImports = await _projectAccessor.OpenProjectForReadAsync(_configuredProject, project =>
         {
             return project
                 .GetItems("Import")
-                .Select(item => item.EvaluatedInclude)
-                .Where(import => !string.IsNullOrEmpty(import))
+                .Select(item => (Value: item.EvaluatedInclude, IsImported: item.IsImported))
+                .Where(import => !string.IsNullOrEmpty(import.Value))
                 .ToImmutableArray();
         });
 
         return existingImports;
     }
     
-    private async Task<string> GetSelectedImportStringAsync()
+    private async Task<IEnumerable<(string Name, string Value)>> GetSelectedImportStringAsync()
```

Comment:
Minor: this method's name says it returns a string, but no longer does.

Rather than rename it to `GetSelectedImportNameValuePairsAsync`, you could keep it returning a string and add the `_encoding.Format` call here instead. That'd remove duplicate calls to that method below too.

Also, you can remove the collection allocation and stream the values using Linq, which the encoding function handles nicely. Something like:

```c#
private async Task<string> GetSelectedImportStringAsync()
{
    ImmutableArray<(string Value, bool IsImported)> existingImports = await GetSelectedImportsAsync();

    return _encoding.Format(
        existingImports
            .Where(pair => !string.IsNullOrEmpty(pair.Value))
            .Select(pair => (Key: pair.Value, Value: pair.IsImported.ToString())));
}
```